textile/features.textile

```diff
@@ -553,9 +554,9 @@ h3(#realtime-connection). Connection
 ** @(RTN17e)@ If the realtime client is connected to a fallback host endpoint, then for the duration that the transport is connected to that host, all HTTP requests, such as history or token requests, should be first attempted to the same datacenter the realtime connection is established with i.e. the same fallback host must be used as the default HTTP request host. If however the HTTP request against that fallback host fails, then the normal fallback host behavior should be followed attempting the request against another fallback host as described in "RSC15":#RSC15
 * @(RTN19)@ Transport state side effects - when a transport is disconnected for any reason:
 ** @(RTN19a)@ Any @ProtocolMessage@ that is awaiting an @ACK@/@NACK@ on the old transport will not receive the @ACK@/@NACK@ on the new transport. The client library must therefore resend any @ProtocolMessage@ that is awaiting a @ACK@/@NACK@ to Ably in order to receive the expected @ACK@/@NACK@ for that message (subject to @RTN7c@/@RTN7d@). The Ably service is responsible for keeping track of messages, ignoring duplicates and responding with suitable @ACK@/@NACK@ messages
```

Comment:
Imagine, connection was in `suspendedState`, after several retries, it's connected to the server ( with new transport ) 
As per `RTN8c` and `RTN9c` we are clearing `connectionKey` and `connectionId`. This means a new connection attempt is not a resume attempt. So, there's neither resume failure nor resume success as a part of a new connection. 

In this case,  what should happen to ack/nack messages? Are we supposed to treat it as a resume failure and send ack/nack messages with incremental message-serial ? Same we do it for channel attach in case of `RTN15g3`

Reply:
Also, it feels we are attaching channels (in attaching, attached or suspended state ) for any type of reconnection 🤔

Reply:
Update - I checked the code, we clear `ack/nack` messages on `suspended` state, so we don't really have to worry about sending them. But can there be a case where a re-connection attempt is not a resume attempt in `disconnected` state?

Reply:
> In this case, what should happen to ack/nack messages?

they'll have been cleared already per RTN7c

> can there be a case where a re-connection attempt is not a resume attempt in disconnected state?

yes: before you're connected for the first time